packages/angular/cli/src/commands/mcp/tools/modernize.ts

```diff
@@ -0,0 +1,192 @@
+      'order listed below, verifying that your code builds and runs between each step:\n\n' +
+      '1. Run `ng g @angular/core:standalone` and select "Convert all components, directives and pipes to standalone"\n' +
+      '2. Run `ng g @angular/core:standalone` and select "Remove unnecessary NgModule classes"\n' +
+      '3. Run `ng g @angular/core:standalone` and select "Bootstrap the project using standalone APIs"',
+    documentationUrl: 'https://angular.dev/reference/migrations/standalone',
+  },
+  {
+    name: 'zoneless',
+    description: 'Migrates the application to be zoneless.',
+    documentationUrl: 'https://angular.dev/guide/zoneless',
+  },
+];
+
+const modernizeInputSchema = z.object({
+  // Casting to [string, ...string[]] since the enum definition requires a nonempty array.
+  transformations: z
+    .array(z.enum(TRANSFORMATIONS.map((t) => t.name) as [string, ...string[]]))
+    .optional(),
+});
+
+export type ModernizeInput = z.infer<typeof modernizeInputSchema>;
+
+export async function runModernization(input: ModernizeInput) {
+  try {
+    if (!input.transformations || input.transformations.length === 0) {
+      const instructions = [
+        'See https://angular.dev/best-practices for Angular best practices. ' +
+          'You can call this tool if you have specific transformation you want to run.',
+      ];
+
+      return {
+        content: [
+          {
+            type: 'text' as const,
+            text: JSON.stringify({
+              instructions,
+            }),
+          },
+        ],
+        structuredContent: {
+          instructions,
+        },
+      };
+    }
+
+    const transformationsToRun = TRANSFORMATIONS.filter((t) =>
+      input.transformations?.includes(t.name),
+    );
+
+    const allInstructions: string[] = [];
+
+    for (const transformation of transformationsToRun) {
+      let transformationInstructions = '';
+      if (transformation.instructions) {
+        transformationInstructions = transformation.instructions;
+      } else {
+        // If no instructions are included, default to running a cli schematic with the transformation name.
+        const command = `ng generate @angular/core:${transformation.name}`;
+        transformationInstructions = `To run the ${transformation.name} migration, execute the following command: \`${command}\`.`;
+      }
+      if (transformation.documentationUrl) {
+        transformationInstructions += `\nFor more information, see ${transformation.documentationUrl}.`;
+      }
+      allInstructions.push(transformationInstructions);
+    }
+
+    const structuredContent = {
+      instructions: allInstructions.length ? allInstructions : undefined,
+    };
+
+    return {
+      content: [{ type: 'text' as const, text: JSON.stringify(structuredContent) }],
+      structuredContent,
+    };
+  } catch (e) {
+    const message = e instanceof Error ? e.message : 'An unknown error occurred.';
```

Comment:
The SDK appears to handle this automatically.  I don't think the try/catch block is needed here.